src/Block/BreadcrumbBlockService.php

```diff
@@ -15,56 +15,52 @@
 
 use Knp\Menu\FactoryInterface;
 use Knp\Menu\ItemInterface;
-use Knp\Menu\Provider\MenuProviderInterface;
 use Sonata\BlockBundle\Block\BlockContextInterface;
 use Sonata\BlockBundle\Meta\Metadata;
+use Sonata\BlockBundle\Meta\MetadataInterface;
 use Sonata\PageBundle\CmsManager\CmsManagerSelectorInterface;
 use Sonata\PageBundle\Model\PageInterface;
 use Sonata\SeoBundle\Block\Breadcrumb\BaseBreadcrumbMenuBlockService;
-use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Twig\Environment;
 
 /**
  * BlockService for homepage breadcrumb.
  *
  * @author Sylvain Deloux <sylvain.deloux@ekino.com>
- *
- * @final since sonata-project/page-bundle 3.26
  */
-class BreadcrumbBlockService extends BaseBreadcrumbMenuBlockService
+final class BreadcrumbBlockService extends BaseBreadcrumbMenuBlockService
 {
-    /**
-     * @var CmsManagerSelectorInterface
-     */
-    protected $cmsSelector;
+    private CmsManagerSelectorInterface $cmsSelector;
 
-    /**
-     * @param string $context
-     * @param string $name
-     */
-    public function __construct($context, $name, EngineInterface $templating, MenuProviderInterface $menuProvider, FactoryInterface $factory, CmsManagerSelectorInterface $cmsSelector)
+    public function __construct(Environment $twig, FactoryInterface $factory, CmsManagerSelectorInterface $cmsSelector)
     {
-        $this->cmsSelector = $cmsSelector;
+        parent::__construct($twig, $factory);
 
-        parent::__construct($context, $name, $templating, $menuProvider, $factory);
+        $this->cmsSelector = $cmsSelector;
     }
 
-    public function getName()
+    public function getName(): string
     {
         return 'sonata.page.block.breadcrumb';
     }
 
-    public function getBlockMetadata($code = null)
+    public function getBlockMetadata($code = null): MetadataInterface
```

Comment:
getBlockMetadata doesn't exist, it's getMetadata now, I think.

Reply:
Hi @VincentLanglet , you are right, the function has been removed. However, there was no message concerning its future deletion. So I didn't dare

Reply:
You have to keep the method, but rename it `getMetadata`

Reply:
there is already the method getMetadata